jgrapht-core/src/main/java/org/jgrapht/graph/interval/IntervalVertex.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * (C) Copyright 2003-2018, by Christoph Grüne and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph.interval;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Implementation for IntervalVertex
+ * This class implements the container class for vertices in an interval graph.
+ *
+ * @param <V> the vertex
+ * @param <T> the interval element
+ *
+ * @author Christoph Grüne (christophgruene)
+ * @since Apr 26, 2018
+ */
+public class IntervalVertex<V, T extends Comparable<T>> implements IntervalVertexInterface, Serializable {
```

Comment:
Calling this class "IntervalVertex" is confusing because it makes it sound like instances are vertices in some graph, which is not the case.  IntervalVertexPair would be good.

Also, I don't really see the need for a separate IntervalVertexInterface; there are no other implementations of this interface, so just use the class directly instead.